Tests/RockLib.Messaging.SQS.Tests/DependencyInjectionTests.cs

```diff
@@ -89,6 +89,27 @@ public void SQSSenderRegisteredClientTest()
             sqsSender.SQSClient.Should().BeSameAs(sqsClient);
         }
 
+        [Fact(DisplayName = "Should register SQS sender as transient")]
+        public void SQSSenderRegisterTransient()
+        {
+            var services = new ServiceCollection();
+
+            using var sqsClient = new AmazonSQSClient(RegionEndpoint.USEast2);
+            services.AddSingleton<IAmazonSQS>(sqsClient);
+
+            services.AddSQSSender("mySender", options =>
+            {
+                options.QueueUrl = new Uri("http://example.com");
+            }, lifetime: ServiceLifetime.Transient);
+
+            var serviceProvider = services.BuildServiceProvider();
+
+            var instance1 = serviceProvider.GetRequiredService<ISender>();
+            var instance2 = serviceProvider.GetRequiredService<ISender>();
+
+            instance1.Should().NotBeSameAs(instance2);
```

Comment:
But **why** should they not be the same? It seems like we should be looking to see that the lifetime is `Transient` in this case for the `ISender` service. Something like:
```csharp
Assert.Contains(services, _ => 
  _.ServiceType == typeof(ISender) && 
  _.Lifetime == ServiceLifetime.Transient);
```
No need to build the service provider. Also, I'd do this with the receiver test as well.

Reply:
They will not be the same because of the the scoping tree, every request will provide a different instance, along with the more important separate instance of `IAmazonSQS`.

But if needed, I can reduce the tests to just look for the registration, I was following the existing pattern.